modules/nw-ingress-gateway-api-deployment-topologies.adoc

```diff
@@ -0,0 +1,135 @@
+// Modules included in the following assemblies:
+//
+// * networking/gateway-api.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-ingress-gateway-api-deployment_{context}"]
+= Gateway API Deployment topologies
+
+Gateway API is designed to accomodate two topologies: shared gateways or dedicated gateways. Each topology has their own advantages and as well as different security implications.
+
+Dedicated gateway:: Routes and any load balancers or proxies are served from the same namespace. The `Gateway`
+object restricts routes to a particular application namespace. This is the default topology for Gateway API in {product-title}.
+
+Shared gateway:: Routes are served from multiple namespaces or with multiple hostnames. The `Gateway` object filters allow routes from application namespaces using the `spec.listeners.allowedRoutes.namespaces` field.
+
+[id="dedicated-gateway-example_{context}"]
+== Dedicated gateway example
+An example of a `Gateway` resource, `fin-gateway`:
+
+.Example dedicated `Gateway` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: fin-gateway
+  namespace: openshift-ingress
+spec:
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 8080
+    hostname: "example.com"
+----
+
+An example of the associated `HTTPRoute` resource, `sales-db`, that attaches to the dedicated `Gateway` object:
+
+.Example `HTTPRoute` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: sales-db
+  namespace: openshift-ingress
```

Comment:
In this example the HTTPRoute `sales-db` is in `openshift-ingress` namespace, but it might not be a case in practice. We mentioned Dedicated gateway previously that  
```
The `Gateway` object restricts routes to a particular application namespace.
```
Maybe we could add namespaces selector to gateway and label to namespace as well (similar to shared gateway but just apply to one dedicated namespace) ?